contracts/Marketplace.sol

```diff
@@ -335,11 +335,19 @@ contract Marketplace is SlotReservations, Proofs, StateRetrieval, Endian {
     _proofReceived(id, proof, pubSignals);
   }
 
-  function markProofAsMissing(SlotId slotId, Period period) public {
-    if (slotState(slotId) != SlotState.Filled)
-      revert Marketplace_SlotNotAcceptingProofs();
+  function canProofBeMarkedAsMissing(
+    SlotId slotId,
+    Period period
+  ) public view slotAcceptsProofs(slotId) {
```

Comment:
I'm not a big fan of view functions that revert. The revert reason is often hard to get to, and it just feels weird that a view on the blockchain state would revert, because there's actually no transaction to revert.

I understand why you did this here, because we already had the logic in `_markProofAsMissing` that reverts when a proof cannot be marked as missing. But I'd rather see a boolean return value on this view function, and no reverts. I'll include a suggestion below as to how this would look in `_markProofAsMissing`.